src/main/kotlin/org/rust/cargo/project/model/impl/CargoProjectImpl.kt

```diff
@@ -314,6 +313,15 @@ data class CargoProjectImpl(
         "CargoProject(manifest = $manifest)"
 }
 
+private inline fun AtomicReference<VirtualFile>.getCachedFile(ioFileGenerator: () -> File?): VirtualFile? {
```

Comment:
Hm, why we parametrize it over a function? Can we give it a plain `path: Path` argument?

BTW, I think one can do someting like `val fooFile: VirtualFile by CachedVirtualFile(fooPath)`.

Reply:
Oh, we've already had the separate class with this code!
But maybe we should use `VirtualFilePointer` instead of `CachedVirtualFile ` indeed? Looks like it does exactly what we need (although there isn't documentation about it)

Reply:
>But maybe we should use VirtualFilePointer instead of CachedVirtualFile indeed? 

Quite probably :-)

Reply:
I've tried to use `VirtualFilePointer` in `CachedVirtualFile` to have convenient syntax with delegating but it requires `Disposable` parent and I've not found good way to pass it - all my attempts looked awful
Let's not change `CachedVirtualFile` now and I'll try to investigate it later

Reply:
>Let's not change CachedVirtualFile now and I'll try to investigate it later

That was my conclusion when I've poked this thing the last time 🤣